osu.Game/Overlays/BeatmapSet/Buttons/FavouriteButton.cs

```diff
@@ -68,6 +69,18 @@ private void load(IAPIProvider api)
                 request?.Cancel();
                 request = new PostBeatmapFavouriteRequest(BeatmapSet.Value?.OnlineBeatmapSetID ?? 0, favourited.Value ? BeatmapFavouriteAction.UnFavourite : BeatmapFavouriteAction.Favourite);
                 request.Success += () => favourited.Value = !favourited.Value;
+                request.Failure += exception =>
+                {
+                    if (exception.Message == "UnprocessableEntity")
```

Comment:
This feels wrong due to hinging on a magic string (which *is* the name of an HTTP error code, but still), and [I don't think 422 isn't guaranteed to be returned on this error only either](https://github.com/ppy/osu-web/blob/666a8135dc576afffb185dd5315d95fb1c4ca302/app/helpers.php#L560-L563). If anything is to be displayed to the user here, it's probably whatever web returns in the `error` message.

Reply:
osu-web should probably be returning an error in the body.

Reply:
surprise, it already was